components/quest-boost/boostCard.tsx

```diff
@@ -1,18 +1,72 @@
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useEffect, useMemo, useState } from "react";
 import styles from "@styles/questboost.module.css";
 import Cardstyles from "@styles/components/card.module.css";
 import Link from "next/link";
 import UnavailableIcon from "@components/UI/iconsComponents/icons/unavailableIcon";
-import { CDNImage } from "@components/cdn/image";
+import CheckIcon from "@components/UI/iconsComponents/icons/checkIcon";
+import TrophyIcon from "@components/UI/iconsComponents/icons/trophyIcon";
+import TokenSymbol from "./TokenSymbol";
+import useBoost from "@hooks/useBoost";
 
 type BoostCardProps = {
   boost: Boost;
+  completedQuests?: number[];
 };
 
-const BoostCard: FunctionComponent<BoostCardProps> = ({ boost }) => {
+const BoostCard: FunctionComponent<BoostCardProps> = ({
+  boost,
+  completedQuests,
+}) => {
+  const [userParticipationStatus, setUserParticipationStatus] =
+    useState<boolean>(false);
+  const [userBoostCheckStatus, setUserBoostCheckStatus] =
+    useState<boolean>(false);
+  const [hasUserCompletedBoost, setHasUserCompletedBoost] =
+    useState<boolean>(false);
+  const { getBoostClaimStatus } = useBoost();
+  const [hovered, setHovered] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!boost || !completedQuests) return;
+    let userParticipationCheck = false;
+    let userBoostCompletionCheck = true;
+    boost.quests.forEach((quest) => {
+      // no quests are completed by user
+      if (!completedQuests) return false;
+
+      // if any of the quests are completed by user and is part of this boost
+      if (completedQuests.includes(quest)) userParticipationCheck = true;
+
+      // check if all quests are completed by the user and if not then set this flag value to false
+      if (!completedQuests.includes(quest)) userBoostCompletionCheck = false;
+    });
+    setHasUserCompletedBoost(userBoostCompletionCheck);
+    setUserParticipationStatus(userParticipationCheck ? true : false);
+  }, [completedQuests]);
+
+  useEffect(() => {
+    if (userParticipationStatus) {
+      const res = getBoostClaimStatus(boost?.id);
+      setUserBoostCheckStatus(res);
+    }
+  }, [userParticipationStatus]);
+
+  const isClickable = useMemo(
+    () => userParticipationStatus && !userBoostCheckStatus,
+    [boost, userParticipationStatus, userBoostCheckStatus]
+  );
+
   return (
-    <Link href={`/quest-boost/${boost?.id}`}>
-      <div className={styles.boost_card_container}>
+    <Link href={isClickable ? `/quest-boost/${boost?.id}` : ""}>
+      <div
+        className={styles.boost_card_container}
+        style={{
```

Comment:
html styling is ugly :/

Reply:
this is needed for handling hover states through js. we would need js checks to enable certain css